File: payment-service/src/main/java/com/example/payment_service/client/RideServiceClient.java
package com.example.payment_service.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.math.BigDecimal;

@FeignClient(name = "ride-service")
public interface RideServiceClient {

    @PutMapping("/api/rides/{rideId}/pay")
    String payRide(
            @PathVariable("rideId") String rideId,
            @RequestParam("amount") BigDecimal amount
    );
}

File: driver-service/src/main/kotlin/com/example/driver_service/client/RideServiceClient.kt
package com.example.driver_service.client

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.*

@FeignClient(    name = "ride-service")
interface RideServiceClient {

    @PutMapping("/api/rides/{rideId}/assign-driver")
    fun assignDriver(
        @PathVariable rideId: String,
        @RequestParam driverId: Long
    )
}